cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -312,46 +312,31 @@ class SimpleArray
     explicit SimpleArray(small_vector<size_t> const & shape,
                          small_vector<size_t> const & stride,
                          std::shared_ptr<buffer_type> const & buffer,
-                         bool is_c_contiguous = true,
-                         bool is_f_contiguous = false)
+                         bool c_contiguous,
+                         bool f_contiguous)
+        : SimpleArray(shape, stride, buffer)
+    {
+        if (shape.size() != stride.size())
+        {
+            throw std::runtime_error("SimpleArray: shape and stride size mismatch");
+        }
+        if (c_contiguous)
+        {
+            check_c_contiguous(shape, stride);
+        }
+        if (f_contiguous)
+        {
+            check_f_contiguous(shape, stride);
+        }
+    }
```

Comment:
@jteng2127 suggest that I could create a constructor with `c_contiguous` and `f_contiguous`, and this constructor could check the contiguous property and call another construcotr without `c_contiguous` and `f_contiguous`.